onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -622,18 +539,15 @@ static void RegisterExecutionProviders(InferenceSession* sess, const std::vector
 #endif
     } else if (type == kCudaExecutionProvider) {
 #ifdef USE_CUDA
-
       auto it = provider_options_map.find(type);
       if (it != provider_options_map.end()) {
-        onnxruntime::CUDAExecutionProviderInfo cuda_provider_options;
-        UpdateCudaProviderOptions(sess, cuda_provider_options, it->second);
-
+        const auto info = CUDAExecutionProviderInfo::FromProviderOptions(it->second);
         RegisterExecutionProvider(
-            sess, *onnxruntime::CreateExecutionProviderFactory_CUDA(cuda_provider_options.device_id,
-                                                                    cuda_provider_options.cudnn_conv_algo,
-                                                                    cuda_provider_options.cuda_mem_limit,
-                                                                    cuda_provider_options.arena_extend_strategy,
-                                                                    cuda_provider_options.do_copy_in_default_stream));
+            sess, *onnxruntime::CreateExecutionProviderFactory_CUDA(info.device_id,
```

Comment:
I know this is not the objective of this PR. Can CUDAExecutionProviderInfo be stored in the factory obj as-is instead of re-create it in CreateProvider() later? This will also keep the signature of this factory function clean - no need to keep adding more parameters as more options are added. Similar comment applies to other EPs like ROCM.

Reply:
ok, i can update the CUDA and ROCM EP factories for now

---
In reply to: [535833627](https://github.com/microsoft/onnxruntime/pull/5923#discussion_r535833627) [](ancestors = 535833627)